File: files/pspro_lin64/examples/c/ShowClient.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

#include "../../h/psprolib.h"

/******************************************************************
*
* This example demonstrates the sgShowClient() function.
* It retrieves information about what the current process
* has checked out.
*
******************************************************************/
int main()
{
 int ret,count;
 char msg[1024],serverhost[256],clienthost[256],user[64],component[20],version[20];

 sgSetAttrInt(SG_ATTR_LICENSETYPE,SG_LICENSETYPE_FLOATING);

 ret = sgCheckout("Monitor","5.0");
 if (ret != SG_SUCCESS) {
   sgGetLastErrorString(msg);
   printf("%s\n",msg);
   exit(0);
 }

 // IMPORTANT:  Even thought the version for "widget" is 6.0 in the license file
 // if we check out an earlier version and only a 6.0 license is available it will
 // consume one of the 6.0 licenses
 ret = sgCheckout("widget","5.0");
 if (ret != SG_SUCCESS) {
   sgGetLastErrorString(msg);
   printf("%s\n",msg);
   exit(0);
 }

 // IMPORTANT!!  if you check out the same component again, the count will increment
 ret = sgCheckout("widget","5.0");
 if (ret != SG_SUCCESS) {
   sgGetLastErrorString(msg);
   printf("%s\n",msg);
   exit(0);
 }

 for(;;) {
     ret = sgShowClient(serverhost,clienthost,user,component,version,&count);
     if (ret == SG_NO_MORE_CLIENT_COMPONENTS) break;

     printf("%s, %s, %s, %s, %s, %d\n",serverhost,clienthost,user,component,version,count);
 }

 sgCheckin("monitor","5.0");

 // NOTE:  Even if you don't check-in the second "widget", it will be 
 // returned to the license pool when the process exits.
 sgCheckin("widget","5.0");

exit(0);
}

File: files/pspro_lin64/examples/fortran/ShowClient.f
      program ShowClient

      include 'psprolib.inc'

!/******************************************************************
!*
!* This example demonstrates the sgShowClient() function.
!* It retrieves information about what the current process
!* has checked out.
!*
!******************************************************************/
      integer retVal,count;
      character*1024 msg
      character*256 serverhost,clienthost
      character*64 user
      character*16 component
      character*12 version

      retVal = sgSetAttrInt(SG_ATTR_LICENSETYPE,SG_LICENSETYPE_FLOATING)

      retVal = sgCheckout("Monitor","5.0");
      if (retVal .ne. SG_SUCCESS) then
        retVal = sgGetLastErrorString(msg)
        write(*,'(A)') msg(:charlen(msg))
        stop
      end if

      ! IMPORTANT:  Even thought the version for "widget" is 6.0 in the license file
      ! if we check out an earlier version and only a 6.0 license is available it will
      ! consume one of the 6.0 licenses

      ret = sgCheckout('widget','5.0')
      if (ret .ne. SG_SUCCESS) then
        retVal = sgGetLastErrorString(msg)
        write(*,'(A)') msg(:charlen(msg))
        stop
      end if

      ! IMPORTANT!!  if you check out the same component again, the count will increment
      ret = sgCheckout('widget','5.0')
      if (ret .ne. SG_SUCCESS) then
        retVal = sgGetLastErrorString(msg)
        write(*,'(A)') msg(:charlen(msg))
        stop
      end if

      do
          retVal = sgShowClient(serverhost,clienthost,user,component,   &
     &                          version,count)
     
          if (retVal .eq. SG_NO_MORE_CLIENT_COMPONENTS) goto 10

          write(*,'(10A,I4)') serverhost(:charlen(serverhost)),' ',     &
     &                       clienthost(:charlen(clienthost)),' ',      &
     &                       user(:charlen(user)),' ',                  &
     &                       component(:charlen(component)),' ',        &
     &                       version(:charlen(version)),' ',            &
     &                       count
 
      end do
  10  continue
     
      retVal = sgCheckin('monitor','5.0')
     
      ! NOTE:  Even if you don't check-in the second "widget", it will be 
      ! returned to the license pool when the process exits.
     
      retVal = sgCheckin('widget','5.0')
      end program
